### DISCUSSION TITLE
Azure Static Web Site yml

### DISCUSSION THREAD

**Discussion by StevenTCramer** [statiqdev/Discussions#43] [Q&A]
I am trying to get our Statiq Blog to deploy to azure static web site.  No lack of effort on number of attempts.

Error message received is:

```console
Failed to find a default file in the app artifacts folder (./Source/Trinsic.Blog/output). Valid default files: index.html,Index.html.
```
> For all the attempt results see https://github.com/trinsic-id/engineering.trinsic.id/actions

The yml file is https://github.com/trinsic-id/engineering.trinsic.id/blob/main/.github/workflows/azure-static-web-apps-yellow-sea-003603510.yml

Their error message uses a relative path and thus I don't know if it is correct.  I do know that index.html file is generated in the directory (./Source/Trinsic.Blog/output) relative to the root of the repo.

I am trying to do same thing as is done in the dotnet foundations site here https://github.com/dotnet-foundation/website/blob/master/.github/workflows/azure-static-web-apps-jolly-grass-05beb181e.yml#L25

That yml does use an older version.  

Anyhelp greatly appreciated.

**Comment by daveaglick** [comment]
Hmm...I don't actually know much about Azure Static Web Sites since I haven't had a chance to play with it yet. At first glance this certainly seems like a mismatch between what the Azure Static Web Sites engine thinks is the root path and where the artifacts are being produced.

Perhaps @clairernovotny has seen this before when setting up the DNF site?

**Comment by StevenTCramer** [comment]
@daveaglick @clairernovotny 

Turns out that `dotnet run` on Ubuntu only generates sitemap.xml yet running it on windows generates all.

Windows:
```
Number of output documents per pipeline and phase:

 | Pipeline          | Input     | Process     | PostProcess | Output    | Total Time |
 |------------------------------------------------------------------------------------|
 | AnalyzeContent    | 5 (0 ms)  |             |             |           | 0 ms       |
 | Archives          |           | 1 (2002 ms) | 1 (5547 ms) | 1 (1 ms)  | 7550 ms    |
 | Assets            |           | 2 (7 ms)    |             | 2 (26 ms) | 33 ms      |
 | Content           |           | 2 (2254 ms) | 2 (5547 ms) | 2 (3 ms)  | 7804 ms    |
 | Data              |           | 0 (11 ms)   |             | 0 (3 ms)  | 14 ms      |
 | DirectoryMetadata | 0 (98 ms) | 0 (2 ms)    |             |           | 100 ms     |
 | Feeds             |           | 0 (0 ms)    |             | 0 (3 ms)  | 3 ms       |
 | Inputs            | 5 (96 ms) | 5 (152 ms)  |             |           | 248 ms     |
 | Redirects         |           | 0 (6 ms)    |             | 0 (3 ms)  | 9 ms       |
 | Sitemap           |           |             | 1 (9 ms)    | 1 (3 ms)  | 12 ms      |
 ```
Ubuntu

```
| Pipeline          | Input     | Process   | PostProcess | Output   | Total Time |
 |---------------------------------------------------------------------------------|
 | AnalyzeContent    | 0 (1 ms)  |           |             |          | 1 ms       |
 | Archives          |           | 0 (0 ms)  | 0 (4 ms)    | 0 (2 ms) | 6 ms       |
 | Assets            |           | 0 (6 ms)  |             | 0 (6 ms) | 12 ms      |
 | Content           |           | 0 (8 ms)  | 0 (4 ms)    | 0 (2 ms) | 14 ms      |
 | Data              |           | 0 (10 ms) |             | 0 (6 ms) | 16 ms      |
 | DirectoryMetadata | 0 (47 ms) | 0 (1 ms)  |             |          | 48 ms      |
 | Feeds             |           | 0 (0 ms)  |             | 0 (6 ms) | 6 ms       |
 | Inputs            | 0 (47 ms) | 0 (27 ms) |             |          | 74 ms      |
 | Redirects         |           | 0 (3 ms)  |             | 0 (6 ms) | 9 ms       |
 | Sitemap           |           |           | 1 (4 ms)    | 1 (8 ms) | 12 ms      |
 ```

**Reply by StevenTCramer** [reply]
Ok the issue is Linux and the wonderful idea of case sensitive file system.

Windows `Input` works Linux wants `input`

So I am betting we need to add 
Directory.GetFiles(path, pattern, new EnumerationOptions { MatchCasing = MatchCasing.CaseInsensitive })
in Statiq

I will see if I can get someone to do a PR.

**Reply by daveaglick** [reply]
That's a tough one... The concern is that on case-sensitive file systems `input` and `Input` actually _are_ different folders and you could conceivably have both. My gut feeling is that Statiq should work like any other app when it comes to the file system - I.e. if I'm running `dotnet run -p [path-to-project]` and my project is in `Foo/Foo.csproj` but I call `dotnet run -p /foo/foo.csproj` in Linux, that wouldn't work either. Maybe a good middle ground, especially with regard to the input folders, is to do a quick scan at the beginning of the run and display a warning if an input folder is specified and doesn't exist, but one with different casing does: https://github.com/statiqdev/Statiq.Framework/issues/169
